### PR TITLE
fix(anta.tests): Updated testcase to support more address family

### LINKED ISSUE #565: Refractor VerifyBGPSpecificPeers and VerifyBGPPeersHealth test to support more bgp address families
Add BGP address family support:
1. path-selection,
2. IPv4/IPv6 SR-TE
3. Link state

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
# Description

Updated VerifyBGPPeerCount, VerifyBGPPeersHealth and VerifyBGPSpecificPeers test to support more address families

Fixes #565 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/custom_types.py [hunk 1]
```diff
@@ -97,8 +97,8 @@ def bgp_multiprotocol_capabilities_abbreviations(value: str) -> str:
     BeforeValidator(interface_autocomplete),
     BeforeValidator(interface_case_sensitivity),
 ]
-Afi = Literal["ipv4", "ipv6", "vpn-ipv4", "vpn-ipv6", "evpn", "rt-membership"]
-Safi = Literal["unicast", "multicast", "labeled-unicast"]
+Afi = Literal["ipv4", "ipv6", "vpn-ipv4", "vpn-ipv6", "evpn", "rt-membership", "path-selection", "link-state", "sr-te"]
+Safi = Literal["unicast", "multicast", "labeled-unicast", "ipv4", "ipv6"]
```
[the remark this answers is not in the document]
Comment (carl-baillargeon) [reply anta/custom_types.py:101]: Let me work on the design today I'll come back to you.

File: anta/tests/routing/bgp.py [hunk 2]
```diff
@@ -290,7 +290,7 @@ def validate_inputs(self: BaseModel) -> BaseModel:
                 """
                 Validate the inputs provided to the BgpAfi class.
 
-                If afi is either ipv4 or ipv6, safi must be provided.
+                If afi is ipv4, ipv6 or sr-te, safi must be provided.
```
Comment (MaheshGSLAB) [inline anta/tests/routing/bgp.py:293]: missed to update, will update back in next commit

File: anta/tests/routing/bgp.py [hunk 3]
```diff
@@ -323,6 +327,10 @@ def test(self) -> None:
 
             afi = cast(Afi, command.params.get("afi"))
             safi = cast(Optional[Safi], command.params.get("safi"))
+
+            # Swaping AFI and SAFI in case of SR-TE address family
+            if afi == "sr-te":
+                afi, safi = safi, afi
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:333]: I think we should put that in the `render` function no?

Comment (MaheshGSLAB) [reply anta/tests/routing/bgp.py:333]: yes we are doing this in render function too in the line (https://github.com/arista-netdevops-community/anta/pull/566/files#:~:text=%23%20For%20SR%2DTE,.vrf)))

Reason of here again to change is to get correct afi and safi in failure. If we dont swap then for SR-TE in failure we get afi as sr-te but expected is ipv4 and safi as sr-te.
```
Failures: [{'afi': 'sr-te', 'safi': 'ipv4', 'vrfs': {'default': {'10.255.0.2': {'peerState': 'Connect', 'inMsgQueue': 0, 'outMsgQueue': 0}}}}]
```

File: anta/tests/routing/bgp.py (hunk 2 above)
Comment (carl-baillargeon) [reply anta/tests/routing/bgp.py:293]: Yes please update this. Also, can we add a note somewhere maybe in the docstring of the test class explaining the different syntax for sr-te?

Comment (MaheshGSLAB) [reply anta/tests/routing/bgp.py]: Updated

File: anta/tests/routing/bgp.py [hunk 4]
```diff
@@ -218,8 +220,12 @@ def render(self, template: AntaTemplate) -> list[AntaCommand]:
         """Render the template for each BGP address family in the input list."""
```
Comment (gmuloc) [inline anta/tests/routing/bgp.py:220]: maybe this is a docstring that deserves more info given the "dark magic" in the rendering function

File: anta/tests/routing/bgp.py [hunk 5]
```diff
@@ -286,7 +298,7 @@ class Input(AntaTest.Input):
         """List of BGP address families (BgpAfi)."""
 
         class BgpAfi(BaseModel):
-            """Model for a BGP address family (AFI) and subsequent service family (SAFI)."""
+            """Model for a BGP address family (AFI) and subsequent address family (SAFI)."""
 
             afi: Afi
```
Comment (gmuloc) [inline anta/tests/routing/bgp.py:303]: Can we work to avoid the duplication of BgpAfi in every test?

**Pull request merged**